Add NewTable method for creating a new Table structure.

smoser opened the pull request:
There was previously no easy way to create an entirely new Table.
The assumption was mostly that some other software had written
a GPT table and this software would read it.

NewTable allows the user to create a new gpt.Table that can be
written to disk with with gpt.Table.Write().

smoser committed:
Add NewTable method for creating a new Table structure.

There was previously no easy way to create an entirely new Table.
The assumption was mostly that some other software had written
a GPT table and this software would read it.

NewTable allows the user to create a new gpt.Table that can be
written to disk with with gpt.Table.Write().

rekby commented on gpt.go:
```diff
@@ -410,6 +410,34 @@ func StringToGuid(guid string) (res [16]byte, err error) {
 
+// NewTable - return a valid empty Table for given sectorSize and diskSize
+//    Note that a Protective MBR is needed for lots of software to read the GPT table.
+func NewTable(sectorSize uint64, diskSize uint64, diskGuid Guid) Table {
```
Hello scott.

Thanks for your patch :)

I suggest use diskSize as first positional arg and struct pointer for all other args - for optional. nil mean defaults.
For example - if guid is empty - generate new random. If sector size is empty - use default 512 byte.

It allow for use set only need args + allow add new args for feature without lose compatible (for example add new GPT version if need, explicit partitions number).

```golang
type NewTableArgs struct {
  SectorSize uint64
  DiskGuid Guid
...
}
func NewTable(diskSize uint64, args *NewTableArgs) Table {
if args == nil {
  args = &NewTableArgs{}
}
if args.SectorSize == 0 {
args.SectorSize = 512
}
...
```

smoser replied:
I didn't want to force the caller to pass a guid, but I also didn't want to add a dependency/import here.  I'll have to add a non-standard-library for uuid generation.

I don't personally feel that this is any more obvious or caller friendly than having the 2 explicit arguments.  It seems that it just "hides" them a little bit.  But I'm fine to let you make the call on that.

smoser replied:
I added a commit adjust to your feedback.
Thanks.

smoser committed:
Adjust call signature of NewTable to take a NewTableArgs.

In order to allow for adding new args in the future without
losing compatibility, change NewTable to take a NewTableArgs
type.  If values in NewTableArgs are un-set, then defaults
will be used, including creation of a new UUID.

smoser committed:
Fix calculation of first usable LBA in NewTable.

Previously the value 34 was just hard coded as the first usable LBA.
That is based on a sector size of 512.  The partitions fit into 5
sectors if the sector size is 4096.  Actually calculate rather than
hard coding.

rekby commented:
Thanks :)

uuidv4 - is only random number, I will do it myself and remove external dependency

rekby commented:
I did :)
a930afbc6edcc89c83d39b79e52025698156178d

smoser commented:
> I did :)
> [a930afb](https://github.com/rekby/gpt/commit/a930afbc6edcc89c83d39b79e52025698156178d)

great thanks.
